### ISSUE TITLE
License filename

### ISSUE THREAD

**Issue by jmid** [c-cube/qcheck#43]
I don't know whether there is a reason for naming the license `.header`. Perhaps this is to include it automatically in some form of generated documentation?

Github however prefers that it is named LICENSE or LICENSE.txt (or something like that) - and more importantly: with this kind of name, the project is automatically highlighted with a logo representing the chosen license. For example, at https://github.com/ocaml/ocaml next to the '116 contributors' there is a little graphics that says LGPL-2.1, making it easy to spot the chosen license.

I am not a lawyer, but the text in .header (modulo the `all rights reserved` and some whitespace issues) seems to match the 2-clause BSD license as presented here: https://opensource.org/licenses/BSD-2-Clause

I (politely) suggest to rename the license file (and perhaps adjust the typesetting to make it recognizable to github) based on the above.

**Comment by jmid** [comment]
Hm. If I press https://github.com/c-cube/qcheck#license on the frontpage it says 
"The code is now released under the LGPL license."

**Comment by c-cube** [comment]
The `.header` file is just for quick inclusion when I create a new file.

**Comment by c-cube** [comment]
I would appreciate if you could find the proper LGPL license file (comes from the temporary inclusion in qtest), add it to the repo, and update header files and `.header`. I'm very busy today ^^

**Comment by jmid** [comment]
Ok, I can try to sort this out. However it seems QCheck has been in a bit of limbo with a license file (.header) signalling a BSD license and a line in the documentation saying LGPL. Hence I would like to settle the matter before just adjusting to LGPL.

So, it seems QCheck has been BSD-licensed at some point. Since iTeML/qtest (which depends on QCheck, and where QCheck lived until ~December 2016) is BSD-licensed (by a similar `.header`) , I suppose we could also stay with that license? That way iTeML/qtest doesn't get into license trouble and QCheck remains compatible with its earlier license.

OCaml has an increasing number of industrial users (Jane Street, Docker, Bloomberg, ...) that all contribute back to the OCaml eco-system. Sticking to a license that is helpful to such users, increases the chance of them using QCheck (and contribute back to it).

Furthermore in the Erlang QuickCheck community it seems the whole license issue has caused a sub-division of QuickCheck users into three:
- Quvic's commercial QuickCheck  http://www.quviq.com/products/  (license unknown to me)
- PropEr https://github.com/manopapad/proper (GPL-3.0)
- Triq https://github.com/krestenkrab/triq (Apache-2.0)

I would really prefer not to sub-divide the relatively limited OCaml QuickCheck community and instead have everyone contribute to make a single, stellar library, licensed as liberally as possible. I'm not sure we can avoid such a division completely as there is already some push for Stephen Dolan's Crowbar QuickCheck/fuzz-testing library (which is MIT licensed). One way to make the case for QCheck is at least to stick to a liberal license. Ultimately it is of course your call as the core author. All of this to say that I believe the issue requires some care (and I think you can tell in which direction I am leaning). :-)

**Comment by jmid** [comment]
Ah, dang. I see now that the source files begin with a LGPL-looking license.

**Comment by c-cube** [comment]
I consider LGPL to be relatively liberal, actually. I would be glad to keep the old BSD license, but since this project spent some time in ITeml/qtest I need to have @vincent-hugot's approval, just to be sure.

**Comment by vincent-hugot** [comment]
I would be okay with BSD.

**Comment by jmid** [comment]
Thanks for taking care of this - and sorry for dropping the ball on the floor! I've changed job (and university) and been through 3 different laptops over the summer, so I'm only now getting an OCaml installation up and running again.

I just noticed that README.adoc still says "The code is now released under the LGPL license." under 'License'. A quick 's/LGPL/BSD' should take care of that though.

Thanks again.
